### ISSUE TITLE
[BUG] No error notification displayed in the UI when join space fails

### ISSUE THREAD

**Issue by dimko** [netdata/netdata-cloud#213]
When a user follows an invitation link to join a space, but joining that space fails for X reason we display a message that she successfully joined that space instead of an error.

e.g 
![image](https://user-images.githubusercontent.com/19268853/145444792-d91b8873-aa2c-45a4-8212-3131018bd27c.png)

**Comment by dimko** [comment]
It seems we need Backend help to finish this one. @burbuli8ra fix works, but there are cases where user successfully joins space but he still gets error message popups.

![image](https://user-images.githubusercontent.com/19268853/150949340-67be2f3c-0504-47a2-96e8-10627021c489.png)

or

![image](https://user-images.githubusercontent.com/19268853/150955860-c3fc45db-697c-46ad-b39e-e380475ce108.png)


The error seems to be coming from the backend.
The location field in the response to /join-callback url is either `https://staging.netdata.cloud/spaces/netdata-ownero-space-hyo1w6e/join-space#error_code=1FYezloCCc-8182213&error_message=space+invitation+not+found&error_msg_key=ErrInvitationNotFound` or `https://staging.netdata.cloud/spaces/netdata-ownero-space-hyo1w6e/join-space#error_code=1FYezloCCc-8182213&error_message=space+invitation+email+mismatch&error_msg_key=ErrEmailMismatch`

**Comment by burbuli8ra** [comment]
Actually @dimko , maybe we can sync offline with @dionysis-netdata on Monday and check if we can merge and close this ticket (FE fix) and open another one for BE investigation and fix implementation.
Just note that if we decide to proceed with my above suggestion, you will have to temporarily disable the failing test in regression suit (maybe with a todo to reenable it when the BE ticket is resolved)... but not quite sure if we want to go that way! 😕

**Comment by dimko** [comment]
@ravilushqa Assigning to you to investigate the BE side

**Comment by dionysis-netdata** [comment]
As per my undestanding this FE fix exposes a **pre-exsiting** bug, which now doesn't appear because of the bug on the FE returning happy messages.

I would image we should release the FE fix and tackle (or accept as low prio) the other bug. 

**Currently** we have 2 bugs:
1. FE returning unreliable error messages
2. BE when joining space

**Comment by dionysis-netdata** [comment]
@ivangergo3 please create a new ticket with what the BE issue

**Comment by ivangergo3** [comment]
I think the issue is coming from the database inconsistency
Let's say one request 

https://testing.netdata.cloud/api/v1/spaces/4a920802-f67a-4f9f-a5bc-ecdd66749497/claimed-nodes?last_updated=2022-03-30T09:54:47.49879923Z

gives back "results": [] an empty array, at the same time 

https://testing.netdata.cloud/api/v1/spaces/4a920802-f67a-4f9f-a5bc-ecdd66749497/rooms/2147de35-50e0-4e6f-a536-c07ce852ea55

response contains details of a node with id f5b9a873-e900-4a7d-aff7-db0ed198d78f
The above node and space Ids should be available in one of the databases, but not all. This is the problem.

**Comment by dionysis-netdata** [comment]
Hmm, can you please create it as ticket and add the "stability" label. There is a task force that is working on dealing with db sync issues

**Comment by ivangergo3** [comment]
Sure, will do it in a second

**Comment by ivangergo3** [comment]
Can we close this? @dionysis-netdata

**Comment by dionysis-netdata** [comment]
> Can we close this? @dionysis-netdata

sure

**Comment by dimko** [comment]
Guys, this isn't fixed. Users fail to join space successfully but we always display the success popup if they got re-directed to space after clicking the invitation email.

**Comment by ivangergo3** [comment]
Blocked by https://github.com/netdata/netdata-cloud/issues/372
The stability team working on https://github.com/netdata/cloud-backend/issues/91
After DBs are being synchronised, we should not see this again.

**Comment by cakrit** [comment]
So we believe that this is a duplicate of #372. Closing it as such and will only reopen after we have removed all materialized views from everywhere.

**Comment by amalkov** [comment]
FYI @shyamvalsan 
@vkuznecovas is still busy with https://github.com/netdata/cloud-backend/issues/91, 
@burbuli8ra we will let you know

**Comment by dimko** [comment]
@shyamvalsan just an update on this one as it got resurrected. The fix by @burbuli8ra works as expected (if invitation fails for some reason, e.g email mismatch, invitation  expired or revoked the user sees an error alert popping up), however the previously mentioned side-effect is still present. After a new user successfully joins and invited space, he doesn't see the correct popup that he successfully joined the space (even though he does), but he sees an error popup that invitation to join failed.

![image](https://user-images.githubusercontent.com/19268853/177132374-fe9713ef-aa49-4b43-83ee-936c3c841467.png)

I propose to not merge the FE fix until this issue is resolved, as this will have a worse effect in UX since it affects the happy path flow and will confuse new users if they have indeed successfully joined the invited space or not.

**Comment by shyamvalsan** [comment]
I agree @dimko - if the change (without the issue being resolved) means that successful joinees will see an error message then best to not release this now.

**Comment by hugovalente-pm** [comment]
this is to be re-evaluated once this one is completed https://github.com/netdata/cloud-frontend/issues/3692

**Comment by burbuli8ra** [comment]
Released on [v2.36.13](https://github.com/netdata/cloud-frontend/releases/tag/2.36.13)
